src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/playback/GerritMissedEventsPlaybackManager.java

```diff
@@ -0,0 +1,486 @@
+
+    private boolean isSupported = false;
+    private boolean playBackComplete = false;
+
+    /**
+     * @param name Gerrit Server Name
+     */
+    public GerritMissedEventsPlaybackManager(String name) {
+        this.serverName = name;
+        checkIfEventsLogPluginSupported();
+    }
+
+    /**
+     * method to verify if plugin is supported
+     */
+    private void checkIfEventsLogPluginSupported() {
+        GerritServer server = PluginImpl.getServer_(serverName);
+        if (server != null && server.getConfig() != null) {
+            isSupported = GerritPluginChecker.isPluginEnabled(
+                    server.getConfig(), EVENTS_LOG_PLUGIN_NAME);
+        }
+    }
+
+    /**
+     * Load in the last-alive Timestamp file.
+     * @throws IOException is we cannot unmarshal.
+     */
+    protected void load() throws IOException {
+        XmlFile xml = getConfigXml(serverName);
+        if (xml.exists()) {
+            serverTimestamp  = (EventTimeSlice)xml.unmarshal(serverTimestamp);
+        }
+    }
+
+    /**
+     * get DateRange from current and last known time.
+     * @return last known timestamp or current date if not found.
+     */
+    protected synchronized Date getDateFromTimestamp() {
+        //get timestamp for server
+        if (serverTimestamp != null) {
+            Date myDate = new Date(serverTimestamp.getTimeSlice());
+            logger.debug("Previous alive timestamp was: {}", myDate);
+            return myDate;
+        }
+        return new Date();
+    }
+
+    /**
+     * When the connection is established, we load in the last-alive
+     * timestamp for this server and try to determine if a time range
+     * exist whereby we missed some events. If so, request the events
+     * from the Gerrit events-log plugin and pump them in to play them back.
+     */
+    @Override
+    public void connectionEstablished() {
+
+        playBackComplete = false;
+        checkIfEventsLogPluginSupported();
+        if (!isSupported) {
+            logger.warn("Playback of missed events not supported for server {}!", serverName);
+            playBackComplete = true;
+            return;
+        }
+        logger.debug("Connection Established!");
+
+        try {
+            load();
+        } catch (IOException e) {
+            logger.error("Failed to load in timestamps for server {}", serverName);
+            logger.error("Exception: {}", e.getMessage(), e);
+            playBackComplete = true;
+            return;
+        }
+        Date timeStampDate = getDateFromTimestamp();
+        long diff = new Date().getTime() - timeStampDate.getTime();
+        if (diff > 0) {
+            logger.debug("Non-zero date range from last-alive timestamp exists for server " + serverName
+                    + ": " + Util.getPastTimeString(diff));
+        } else {
+            logger.debug("Zero date range from last-alive timestamp for server " + serverName);
```

Comment:
logger.debug("Zero date range from last-alive timestamp for server {}", serverName);